index.js

```diff
@@ -200,173 +215,329 @@ while (args.length) {
+			outputError("PAGE ERROR", err); 
+		});
 
-	if (matchMQ) {
-		options.matchMQ = true;
-	}
+		page.on("error", function (err) {  
+			outputError("PAGE ERROR", err);
+		});
+
+		if(!await loadPage() || !await injectCssExtractor()) {
+			await closePuppeteer();
+		}
 
-	if (required) {
-		options.required = required;
+	}
+	catch (e) {
+		outputError("EXCEPTION", e);
+		try {
+			await closePuppeteer();
+		} catch (err) {}
 	}
 
-	if (localStorage) {
-		page.evaluate(function (data) {
-			var storage = window.localStorage;
-			if (storage) {
-				for (var key in data) {
-					storage.setItem(key, data[key]);
-				}
+	return;
+
+	async function loadPage() {
+
+		if (url) {
+	
+			debug.loadTime = new Date();
+	
+			await page.goto(url);
+
+			return true;
+		}
+		else {
+		
+			if (!fakeUrl) {
+				stderr("Missing \"fake-url\" option");
+				return false;
 			}
-		}, localStorage);
-	}
+		
+			html = fs.readFileSync(0, "utf-8");
+		
+			debug.loadTime = new Date();
+	
+			await page.setRequestInterception(true);
+	
+			page.once("request", req => {
+			  req.respond({
+				body: "<!DOCTYPE html><html><head><title>Empty page</title></head><body><div>Empty page</div></body></html>"
+			  });
+			});
+	
+			await page.goto(fakeUrl);
+	
+			if (!stripResources) {
+				// disable request interception to allow caching
+				await page.setRequestInterception(false);
+			}
+	
+			if (diskCacheDir) {
+				await page.setCacheEnabled(true);
+			}
+	
+			await page.setContent(html);
 
-	if (Object.keys(options).length) {
-		page.evaluate(function (options) {
-			window.extractCSSOptions = options;
-		}, options);
+			return true;
+		}	
 	}
 
-	if (!height) {
-		var _height = page.evaluate(function () {
-			return document.body.offsetHeight;
-		});
-		page.viewportSize = {
-			width: width,
-			height: _height
-		};
-	}
+	async function injectCssExtractor() {
 
-	if (!fs.isFile(scriptPath)) {
-		fail("Unable to locate script at: " + scriptPath);
-	}
+		if (!html) {
+			html = await page.content();
+		}
 
-	var injection = page.injectJs(scriptPath);
-	if (!injection) {
-		fail("Unable to inject script in page");
-	}
+		if(html.indexOf("stylesheet") === -1) {
+			return false;
+		}
 
-};
+		debug.loadTime = new Date() - debug.loadTime;
+	
+		var options = {};
+	
+		if (matchMQ) {
+			options.matchMQ = true;
+		}
+	
+		if (required) {
+			options.required = required;
+		}
+	
+		if (localStorage) {
+			await page.evaluate(function (data) {
+				var storage = window.localStorage;
+				if (storage) {
+					for (var key in data) {
+						storage.setItem(key, data[key]);
+					}
+				}
+			}, localStorage);
+		}
+	
+		if (Object.keys(options).length) {
+			await page.evaluate(function (options) {
+				window.extractCSSOptions = options;
+			}, options);
+		}
+	
+		if (!height) {
+			var _height = await page.evaluate(function () {
+				return document.body.offsetHeight;
+			});
+			page.viewportSize = {
+				width: width,
+				height: _height
+			};
```

Comment:
This also needs to be updated to use the puppeteer api:

```js
page.setViewport({
	width: width,
	height: _height
});
```

Reply:
Nice find, it is now fixed.